app/group.go

```diff
@@ -86,30 +88,51 @@ func (a *App) UpsertGroupSyncable(groupSyncable *model.GroupSyncable) (*model.Gr
 		return nil, err
 	}
 
-	if gs == nil {
-		gs, err = a.Srv().Store.Group().CreateGroupSyncable(groupSyncable)
+	// reject the syncable creation if the group isn't already associated to the parent team
+	if groupSyncable.Type == model.GroupSyncableTypeChannel {
+		var channel *model.Channel
+		channel, err = a.Srv().Store.Channel().Get(groupSyncable.SyncableId, true)
```

Comment:
Sorry, I'm not sure if I can review this section of the code because I don't really understand what's happening. :)